### PR TITLE
Arbitrarymarkdown

### LINKED ISSUE #938: Implement RenderArbitraryMarkdown method
I checked IMiscellaneousClient.cs, it seems that currently we only have the method that renders raw markdown. Shall we implement the method to render arbitrary markdown? 
https://developer.github.com/v3/markdown/#render-an-arbitrary-markdown-document

### PR DESCRIPTION
**Pull request by chenjiaming93**
Let us see if #904 still happens

Fixes #938 
Supersedes #939

### CODE REVIEW

File: Octokit.Tests.Integration/Clients/PullRequestsClientTests.cs [hunk 1]
```diff
@@ -37,7 +37,6 @@ public PullRequestsClientTests()
         var result = await _fixture.Create(Helper.UserName, _context.RepositoryName, newPullRequest);
 
         Assert.Equal("a pull request", result.Title);
-        Assert.False(result.Merged);
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/PullRequestsClientTests.cs]: This seems unrelated to the other changes - could you add it back in?

File: Octokit.Tests/Clients/MiscellaneousClientTests.cs [hunk 2]
```diff
@@ -31,7 +31,27 @@ public class TheRenderRawMarkdownMethod
                     "text/plain");
             }
         }
+        public class TheRenderArbitrryMarkdownMethod
```
Comment (shiftkey) [inline Octokit.Tests/Clients/MiscellaneousClientTests.cs]: typo: `Arbitrry` -> `Arbitrary`

File: Octokit.Tests/Clients/MiscellaneousClientTests.cs [hunk 3]
```diff
@@ -31,7 +31,27 @@ public class TheRenderRawMarkdownMethod
...
+        {
+            [Fact]
+            public async Task RequestsTheEmojiEndpoint()
```
Comment (shiftkey) [inline Octokit.Tests/Clients/MiscellaneousClientTests.cs]: This test description should mention that it's interested in the markdown endpoint.

**Comment by shiftkey** [comment]
There seems to be some changes in here related to the issue search. Could you refresh my memory on which issue those changes are addressing?

**Comment by chenjiaming93** [comment]
@shiftkey 
Those are about #839

**Comment by shiftkey** [comment]
@chenjiaming93 

Given there's some other branches for Pull Request Assignee and Pull Request Merged, you could have a go at rewriting your branch to contain just those commits.

I'll walk you through the process to explain what's happening. You should have a remote named `origin` and one names `upstream`, you can check them like this:

> git remote -vv

If you only see `origin` listed, run this:

> git remote add upstream https://github.com/octokit/octokit.net.git

And now you've done that, ensure you've got the latest changes from both repositories:

> git fetch --all

We'll switch to our branch we're working on:

> git checkout arbitrarymarkdown

And we're going to put it back to whatever is the latest from upstream:

> git reset upstream/master --hard

If at this point you feel unsure about continuing, this is how you get back to your previous state:

> git reset origin/arbitrarymarkdown --hard

Now, we're going to add those three commits onto our new branch:

> git cherry-pick cde7254d
> git cherry-pick 7c1c86fc9
> git cherry-pick b084d9f5

Once you've tested out things with a `.\build BuildApp`, you can push this to reset the PR:

> git push origin arbitrarymarkdown --force

**Comment by chenjiaming93** [comment]
@shiftkey  no problem.  I will do that. Thx

**Comment by shiftkey** [comment]
Closing in favour of #957
